backend/functions/student/functions.js

```diff
@@ -37,17 +37,32 @@ const addStudentSurveyResponse = async (
     courseCatalogNames.map((name) => mapCatalogNameToCrseId(name))
   );
 
-  const studentUpdate = studentRef
+  const existingData = (await studentRef.doc(email).get()).data();
+  //studentCourses becomes courseIds of existingData.groups if available, otherwise []
+  let studentCrses = existingData ? existingData.groups : [];
+  const studentCrseIds = studentCrses.map((crse)=> crse.courseId);
+  let crsesToAdd = new Array();
+  let crseCatalogsToUpdate = new Array();
+  crseIds.forEach((crse, index) => {
+    if (!studentCrseIds.includes(crse)) {
+      crsesToAdd.push(crse);
+      crseCatalogsToUpdate.push(courseCatalogNames[index]);
+    }
+  })
+
+  crsesToAdd.forEach((crse) => studentCrses.push({
+    courseId: crse,
+    groupNumber:-1
+  }))
+
+  const studentUpdate = await studentRef
     .doc(email)
     .set({
       name,
       college,
       year,
       preferredWorkingTime,
-      groups: crseIds.map((crseId) => ({
-        courseId: crseId,
-        groupNumber: -1,
-      })),
+      groups: studentCrses
```

Comment:
This pull request looks great! I like how you condensed this block since you already know what each mapping will return ahead of time. The variable names are also well chosen to be self-explanatory, so great job on that!